### ISSUE TITLE
Working with Shared Drives

### ISSUE THREAD

**Issue by rjshreve** [masbug/flysystem-google-drive-ext#57]
Hello - I'm thinking this is less of an issue and more of just me not understanding but I'm trying to work within Shared Drives and not having much luck. The structure I'm going for is:

Shared Drive
- Sub Folder 1
- - File 1
- Sub Folder 2
- - File 1
- - File 2
I've been able to use `Storage::disk('google')->makeDirectory('Sub Folder 1')` and that works fine. The problem comes when I try to use `Storage::disk('google')->put('Sub Folder 1/File 1.txt', 'Hello World');` - In this case it ends up creating a "Sub Folder 1" and then I get the error in saying "Unable to read file from location: [shared drive id]/Sub Folder 1. File not found" and the folder it created is empty. 

Essentially, I can’t do anything within folder inside a shared drive. I can only perform actions in the root of shared drive.

Thanks in advance!

**Comment by rjshreve** [comment]
For the record, this is cross posted. I asked @erikn69 as well on their repo but we weren’t able to find a solve.

**Comment by erikn69** [comment]
But wich laravel versión? Package versión?

**Comment by rjshreve** [comment]
Apologies! Laravel 9, Flysystem 3

**Comment by erikn69** [comment]
I already test team drives, and shared folders on Laravel 9, Flysystem 3, everything working as expected

**Comment by xel1045** [comment]
I have the same problem on my side and it seems to be related with the `setPathPrefix` in method `setTeamDriveId`. When I update it to the following, `listContents` works just fine:

```php
    public function setTeamDriveId($teamDriveId, $corpora = 'teamDrive')
    {
        $this->enableTeamDriveSupport();
        $this->optParams = array_merge_recursive($this->optParams, [
            'files.list' => [
                'corpora' => $corpora,
                'includeTeamDriveItems' => true,
                'teamDriveId' => $teamDriveId
            ]
        ]);

        if ($this->root === 'root' || $this->root === null) {
            $this->setPathPrefix(''); // $this->setPathPrefix($teamDriveId);
            $this->root = $teamDriveId;
        }
    }
```

Also, I see that the parameter `includeTeamDriveItems` is deprecated and should be replace by `includeItemsFromAllDrives` (see https://developers.google.com/drive/api/v3/reference/files/list)

**Comment by erikn69** [comment]
@xel1045 run the tests, with and without your change
(be carefull with the `usePermanentDelete`)

**Comment by masterix21** [comment]
@xel1045, thanks: it works for me.

**Comment by pablomalo-cooperl** [comment]
I have the exact same problem as @rjshreve. The fix proposed by @xel1045 works, thank you for the suggestion. Where do we go from here, @erikn69? Would it be appropriate to open a PR?

**Comment by masterix21** [comment]
Who needs a fix could apply the @xel1045's solution using the `spatie/invade` package like so:
```php
$adapter = new GoogleDriveAdapter(
    service: new \Google_Service_Drive($client),
    root: '',
    options: [ 'teamDriveId' => $config['teamDriveId'] ]
);

invade($adapter)->setPathPrefix('');
```

**Comment by pablomalo-cooperl** [comment]
@masterix21 Yes but that is more of a hack than a fix, right? 

Edit: Also, though I recognize the potential usefulness of `spatie/invade` in working around situations like these, I really feel it is bad practice to trespass on private properties (pun intended).

**Comment by masterix21** [comment]
Yes, it is a hack.

**Comment by erikn69** [comment]
Fixed on #63, please test that branch
Thanks @xel1045 for the suggestions

**Comment by xel1045** [comment]
Good news, I'm really glad it helped!
